docs/source/migrate/index.rst

```diff
@@ -0,0 +1,274 @@
+    where
+      signatory p
+
+The package ``foo-2.0.0`` contains exactly the same modules, but a new template ``Bar`` has been
+added to the ``Foo`` module.
+
+.. code-block:: daml
+
+  daml 1.2
+  module Foo where
+
+  template Foo
+    with
+      a : Int
+      p : Party
+    where
+      signatory p
+
+  template Bar
+    with
+      t : Text
+      p : Party
+    where
+      signatory p
+
+
+If we generate a migration project with ``daml migrate`` as above, the directory contents of the
+``foo-2.0.0-upgrade/daml`` directory is
+
+.. code-block:: shell
+
+  daml
+  ├── FooAInstances.daml
+  ├── FooBInstances.daml
+  └── Foo.daml
+
+For every template that was defined in the module ``Foo`` in the ``foo-1.0.0`` package, you will
+find two new templates. One to upgrade contract instances of this template to the changed template
+defined in the module ``Foo`` in the package ``foo-2.0.0`` and one to rollback the process. Here is
+an example:
+
+.. code-block:: daml
+
+   template FooUpgrade
+    with
+        op : Party
+    where
+        signatory op
+        nonconsuming choice Upgrade: ContractId B.Foo
+            with
+                inC : ContractId A.Foo
+                sigs : [Party]
+            controller sigs
+                do
+                    d <- fetch inC
+                    assert $ fromList sigs == fromList (signatory d)
+                    create $ conv d
+
+   template FooRollback
+    with
+        op : Party
+    where
+        signatory op
+        nonconsuming choice Rollback: ContractId A.Foo
+            with
+                inC : ContractId B.Foo
+                sigs : [Party]
+            controller sigs
+                do
+                    d <- fetch inC
+                    assert $ fromList sigs == fromList (signatory d)
+                    create $ conv d
+
+Currently, we only support one upgrade/rollback contract template pair. In the future we might
+extend the ``migrate`` command with different possible upgrade/rollback templates. The above
+contract template offers a choice to input a contract instance defined in ``foo-1.0.0``, convert it
+with the ``conv`` function and create a new one that follows the data type definition in
+``foo-2.0.0``. The heart of the migration code is the ``conv`` function. It is defined in the DAML
+standard library in the module ``DA.Upgrade`` and has the following type signature:
+
+.. code-block:: daml
+
+  conv : (Generic a repA, Generic b repB, Conv repA repB) => a -> b
+
+From its type signature you can see that it can convert any two data types that are instances of the
+``Generic`` class and whose generic representation can be converted itself.
+
+The ``migrate`` command takes care of defining generic instances for all relevant data types in the
+two packages. In our example, you'll find them in the files ``FooAInstances.daml`` and
+``FooBInstances.daml``.
+
+Generic representations can be converted when they are ``isomorphic``. That means the corresponding
```

Comment:
Is there  are function called `isomorphic`? If not, we should change the formatting from code-like to italic.

Reply:
changed.